backend/src/ml_space_lambda/dataset/lambda_functions.py

```diff
@@ -305,23 +305,32 @@ def create_dataset(event, context):
 @api_wrapper
 def list_resources(event, context):
     username = event["requestContext"]["authorizer"]["principalId"]
+    is_admin = event["requestContext"].get("resourcePath") == "/admin/datasets"
```

Comment:
This is perfectly valid but it feels a little something to be basing if they're an admin on the URL. Especially with how hard to follow our authorizer is. Not blocking but I'd suggest this be changed to closer to how it is done [in a few other places](https://github.com/awslabs/mlspace/blob/develop/backend/src/ml_space_lambda/group/lambda_functions.py#L67-L69).

```
    user = UserModel.from_dict(json.loads(event["requestContext"]["authorizer"]["user"]))
    if Permission.ADMIN in user.permissions:
```

Reply:
This is a good point; I think I'll combine the two. I don't want to return all datasets every time an admin tries to list datasets, only when they're listing datasets from the admin datasets page (which uses the /admin/datasets endpoint)